packages/flags/src/utils.ts

```diff
@@ -0,0 +1,44 @@
+import { Flags, Flag } from './index';
+
+const FF = 'ASAP_FF';
+
+export const mergeFlags = (flags: Flags): Partial<Record<Flag, boolean>> => {
+  if (process.browser) {
+    return document.cookie
+      .split(';')
+      .reduce<Record<string, boolean>>((acc, cookie) => {
+        const [name, value] = cookie.split('=');
+        if (isFF(name, value, flags)) {
+          Object.entries(JSON.parse(value)).forEach(([flag, val]) => {
+            if (Object.keys(flags).includes(flag)) {
+              acc[flag] = val as boolean;
+            }
+            return acc;
+          });
+        }
+
+        return acc;
+      }, flags);
+  }
+  return flags;
+};
+
+export const isValidFF = (str: string, flags: Flags) => {
+  try {
+    const parsedValue = JSON.parse(str);
+
+    if (Array.isArray(parsedValue)) return false;
+
+    if (Object.keys(parsedValue).length === 0) return false;
+
+    return Object.entries(parsedValue).some(
+      ([key, val]) =>
```

Comment:
I think that it's ok having additional keys on the cookie value as soon as we don't add them as feature flags. 
What do you think?